src/prefect/agent/docker/agent.py

```diff
@@ -399,17 +415,25 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
             host_config.update(extra_hosts={"host.docker.internal": docker_internal_ip})
 
         networking_config = None
+        # At the time of creation, you can only connect a container to a single network,
+        # however you can create more connections after creation.
+        # Connect first network in the creation step. If no network is connected here the container
+        # is connected to the default `bridge` network.
+        # The rest of the networks are connected after creation.
+        if self.networks:
+            networking_config = self.docker_client.create_networking_config(
+                {self.networks[0]: self.docker_client.create_endpoint_config()}
+            )
```

Comment:
Interestingly it is not possible to add multiple networks while creating the container. The docker-py documentation mentions under "Networking":

>At the time of creation, you can only connect a container to a single networking, but you can create more connections by using connect_container_to_network().

Reply:
Note that adding all networks via `connect_container_to_network` below results in slightly different behaviour where also the default `bridge` network is added to the container (seemingly as a default when no `networking_config` is passed to `create_container`.

Reply:
I don't think this should be a problem. Is there a case where you wouldn't want to default bridge network?

Reply:
I don't know of a case where you wouldn't want to connect to the default network.
However not adding the `bridge` network as a default has the advantage that:
- It is consistent with previous behaviour of just passing in a single network.
- Still allows to pass in the bridge network explicitly.

That being said, I don't have a strong personal preference and happy to implement either here.

Reply:
I think having bridge added by default should be fine and we can adjust this behavior later if someone needs it to function otherwise 👍